### PR TITLE
MEN-2961: acceptance tests for workflows

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

File: tests/docker-compose.acceptance.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+version: '2.1'
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply tests/docker-compose.acceptance.yaml]: @lluiscampos ok, renaming the file to use the `yml` extension. The docker-compose in the root folder is there to actually run mongodb and, optionally, the server and worker for manual testing. `workflows` can work as a stand-alone tool, without the whole `integration` from Mender. I used it for demoing several weeks ago.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -19,12 +19,57 @@ before_script:
...
+    - docker load -i prtest_image.tar
+    - docker load -i prtest_worker_image.tar
+    - docker load -i acceptance_testing_image.tar
+  script:
+    - make acceptance-testing-up
+    - sh -c "sleep 30"
+    - make acceptance-testing-run
+  after_script:
+    - make acceptance-testing-logs
+    - make acceptance-testing-down
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply .gitlab-ci.yml]: @lluiscampos sounds a wonderful approach, thanks.

File: tests/tests/test_provision_device.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        res = requests.get(
+            workflows_url + "/api/v1/workflow/provision_device/" + response["id"]
+        )
+        assert res.status_code == 200
+        # if status is done, break
+        response = res.json()
+        assert response is not None
+        assert type(response) is dict
+        if response["status"] == "done":
+            break
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply tests/tests/test_provision_device.py]: @lluiscampos getting the "inprogress" is tricky, as latency is very low (below 20ms). I think it doesn't make sense to test it.

File: tests/tests/test_provision_device.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    assert res.status_code == 201
+    # verify the response
+    response = res.json()
+    assert response is not None
+    assert type(response) is dict
+    assert response["name"] == "provision_device"
+    assert response["id"] is not None
+    # get the job details, every second until done
+    for i in range(10):
+        time.sleep(1)
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply tests/tests/test_provision_device.py]: yes, it actually completes in tens of ms, which makes tricky to catch the "in-progress". On my workstation I get it half of the times, and this makes the test too fragile, IMHO.

File: tests/tests/test_provision_device.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lluiscampos) [reply tests/tests/test_provision_device.py]: Ok, I trust the expert here.

**Comment by tranchitella** [comment]
> I run "make" locally and got a linting error, running with go version 1.13.8.
> -> My point being is that the linting from the local Makefile is different that the "static checks" that we do fro mendertesting template (as these seem to pass in CI). I suggest we add a `make lint` in this pipeline or we remove it from the Makefile, as it is not nice to have a non-functional default make target.
> The error itself is:
> 
> ```
> /home/lluis/go/src/github.com/mendersoftware/workflows/store/mongo/datastore_mongo.go:360:17: should omit type *model.Job from declaration of var aquiredJob; it will be inferred from the right-hand side
> ```
> 
> Other than that, see below my individual comments for small adjustments and discussions.

Interestingly enough, I cannot get this error locally on MacOS. I've upgrade my golang to 1.13.8, and I still don't get this error: lint and vet are happy here. Any idea why this happens?

**Comment by lluiscampos** [comment]
> Interestingly enough, I cannot get this error locally on MacOS. I've upgrade my golang to 1.13.8, and I still don't get this error: lint and vet are happy here. Any idea why this happens?

I updated my `golint` and the error disappeared, so the problem was local as I was using an outdated `golint`. Sorry for the noise; I don't have an strong opinion anymore in modifying the pipeline to add this check.

**Comment by tranchitella** [comment]
@lluiscampos merging it as it is, as it unlocks @merlin-northern to add additional workflows & acceptance tests. I believe it makes sense to add additional make targets when we agree on a common schema we can apply to all our packages.

**Pull request merged**